src/webpack/plugin/SitePlugin.js

```diff
@@ -0,0 +1,287 @@
+        use: [
+          babelLoader,
+          'devSiteExample',
+        ],
+      }, {
+        test: /\.json$/,
+        // this bypasses the default json loader
+        type: 'javascript/auto',
+        resourceQuery: '?dev-site-package',
+        use: [
+          babelLoader,
+          'devSitePackage',
+        ],
+      }, {
+        resourceQuery: '?dev-site-props-table',
+        use: [
+          babelLoader,
+          {
+            loader: 'devSitePropsTable',
+            options: {
+              mdx: mdxOptions(compiler.options.output.publicPath),
+              resolveExtensions: compiler.options.resolve.extensions,
+            },
+          },
+        ],
+      },
+      // Spread the original loaders. These will be applied if all above loaders fail.
+      ...compiler.options.module.rules,
+      ],
+    }];
+
+    // RESOLVE
+    // If plugins is not defined, define it.
+    if (!compiler.options.resolve.plugins) {
+      compiler.options.resolve.plugins = [];
+    }
+
+    // If a mono repo, update the rootDirectories to include all the packages.
+    const rootDirectories = [
+      ...isLernaMonoRepo ? [path.resolve(processPath, 'packages', '*')] : [processPath],
+    ];
+
+    // Switch between source and distribution files.
+    compiler.options.resolve.plugins.push(
+      new DirectorySwitcherPlugin({
+        shouldSwitch: compiler.options.mode !== 'production',
+        source: sourceFolder,
+        distribution: distributionFolder,
+        rootDirectories,
+      }),
+    );
+
+    // Alias the local package to allow imports to reference the file as if it was imported from node modules.
+    compiler.options.resolve.plugins.push(
+      new LocalPackageAliasPlugin({
+        rootDirectories,
+      }),
+    );
+
+    // RESOLVE LOADER
+    // add the path to search for dev site loaders
+    compiler.options.resolveLoader.modules = [
+      path.resolve(__dirname, '..', 'loaders'),
+      'node_modules',
+    ];
+
+    // generate the 404 page.
+    new HtmlWebpackPlugin({
+      filename: '404.html',
+      template: path.join(__dirname, '..', 'templates', '404.html'),
+      inject: 'head',
+      chunks: ['redirect'],
+    }).apply(compiler);
+
+    // WEBPACK DEV SERVER
+    if (compiler.options.devServer) {
+      // Setting this to enable browser routing
+      compiler.options.devServer.historyApiFallback = true;
+    }
+
+    new DefinePlugin({
+      // Base name is used to namespace terra-dev-site this is used in redirect.js which is only used in the 404 page.
+      TERRA_DEV_SITE_BASENAME: JSON.stringify(basename),
+    }).apply(compiler);
+  }
+
+  apply(compiler) {
+    // Use default public path else the env else /
+    let defaultPublicPath;
+    if (compiler.options.output && compiler.options.output.publicPath) {
+      defaultPublicPath = compiler.options.output.publicPath;
+    }
+    const publicPath = defaultPublicPath || process.env.TERRA_DEV_SITE_PUBLIC_PATH || '/';
```

Comment:
Why is the webpack config's publicPath honored before `TERRA_DEV_SITE_PUBLIC_PATH`?

Reply:
Good question, the env should probably override everything